### PR TITLE
feat: autocomplete manager for both TagTypes and TaxonomyNames

### PR DESCRIPTION
**Pull request by monsieurtanuki**
### What
- Now we have a common autocomplete manager for `TagType` ("old" taxonomies) and `TaxonomyName` ("elastic search" taxonomies).
- This manager is needed because while the user is typing, the server response may happen in the wrong order, time-wise.

### Part of 
- https://github.com/openfoodfacts/smooth-app/pull/4871
- https://github.com/openfoodfacts/smooth-app/issues/4068

### Files
New files:
* `autocomplete_manager.dart`: Manager that returns the suggestions for the latest input. Code was largely moved from `suggestion_manager.dart` as refactoring. Manager that returns the suggestions for the latest input.
* `autocompleter.dart`: Interface that provides autocomplete suggestions.
* `tag_type_autocompleter.dart`: Autocomplete suggestions for TagType.
* `taxonomy_name_autocompleter.dart`: Autocomplete suggestions for TaxonomyNames.

Impacted files:
* `api_suggestion_manager_test.dart`: added test for elastic search
* `fuzziness.dart`: renamed from `fuzziness_level.dart`
* `open_food_search_api_client.dart`: minor refactoring
* `openfoodfacts.dart`: added the new 4 files and impacted the file renaming
* `suggestion_manager.dart`: deprecated it in favor of new classes `TagTypeAutocompleter` and `AutocompleteManager`

### CODE REVIEW

File: lib/src/utils/autocomplete_manager.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+/// Manager that returns the suggestions for the latest input.
+///
+/// Typical use case: the user types one character (which triggers a call to
+/// suggestions), then another character (which triggers another call).
+/// What if the second call is much faster than the first one, for server or
+/// connection reasons? The autocomplete widget will get the second suggestions,
+/// then the first suggestions will override them.
+/// And the user should get the suggestions that match the latest input.
+class AutocompleteManager implements Autocompleter {
```
Comment (g123k) [inline lib/src/utils/autocomplete_manager.dart:14]: I wonder if you shouldn't provide a `clear` method, to allow the client to clear the cache, in the case where the same Manager is used for multiple queries

**Review by g123k — COMMENTED** [review]
Just a minor comment, otherwise 👌

**Review by M123-dev — APPROVED** [review]
I've waited to review it since there was an unanswered comment from Edouard. But let's not keep this stale even longer. 

Autocomplete is nice to have and the code looks good, thanks @monsieurtanuki

**Comment by monsieurtanuki** [comment]
Thank you @M123-dev for your review!

**Pull request merged**